### DISCUSSION TITLE
Analogue output question

### DISCUSSION THREAD

**Discussion by pipcoen** [harp-tech/protocol#79] [Q&A]
I wanted to clarify a point: if I need multiple Analogue outputs (say 3 or 4), how many are possible with the output expander, and at what sampling rate?

**Comment by filcarv** [comment]
The outputs of the Harp Output Expander are all digital, they don't have analog capabilities.

**Reply by pipcoen** [reply]
Thanks for getting back to me. So to clarify, is there any way to increase the number of analogue inputs/outputs?

**Reply by filcarv** [reply]
You can increase the number of channels by using the Harp Expander Breakout (photo below).
If plugged to the Harp Output Expander, it gives 5 additional digital outputs and 2 digital inputs.

The Harp Expander Output doesn't have analog capabilities, it's a digital device. Nevertheless, it has an expansion port (where the Harp Expander Breakout is plugged) that can be used to plug peripherals and expand the features. These peripherals need to be developed according to specific specs, that can be analog if needed.

The current available Peripherals are a magnetic rotary encoder, a servo driver controller, and the breakout.

![harp_expander_and_breakout](https://user-images.githubusercontent.com/34546039/219671678-df8abd34-14eb-420e-affc-24dfc8f58d95.jpg)
